src/lib.rs

```diff
@@ -459,6 +461,36 @@ pub trait Rng: RngCore {
         T::Sampler::sample_single(low, high, self)
     }
 
+    /// Generate a random value in the range `[0, high)`, i.e. inclusive of
+    /// `0` and exclusive of `high`.
+    ///
+    /// This function is optimised for the case that only a single sample is
+    /// made from the given range. See also the [`Uniform`] distribution
+    /// type which may be faster if sampling from the same range repeatedly.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `high` is `0` or negative.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rand::{thread_rng, Rng};
+    ///
+    /// let mut rng = thread_rng();
+    /// let n: u32 = rng.gen_below(10);
+    /// println!("{}", n);
+    /// let m: f64 = rng.gen_below(123.45f64);
+    /// println!("{}", m);
+    /// ```
+    ///
+    /// [`Uniform`]: distributions/uniform/struct.Uniform.html
+    fn gen_below<T: SampleUniform, B>(&mut self, high: B) -> T
+        where B: SampleBorrow<T> + Sized,
+              T: Default {
```

Comment:
Myself I prefer using `T: From<u8>` and `0.into()` / `from(0)`. It works for all the common types and it's explicitly zero, but functionally there's no difference it seems (not sure if it's required that `default() == 0`).

Reply:
I used `Default` because it's a very commonly implemented trait, and one that's implementable using `#[derive]`. For number-ish types, I couldn't see it ever implemented as anything other than 0.

More concretely it's implemented by `Duration` which `From<u8>` isn't.

Reply:
Also seeing that SIMD types implement `Default` but not `From<u8>`.